auth/common.cc

```diff
@@ -40,7 +40,8 @@ static logging::logger auth_log("auth");
 
 bool legacy_mode(cql3::query_processor& qp) {
     return !qp.db().get_config().check_experimental(
-        db::experimental_features_t::feature::AUTH_V2);
```

Comment:
`auth: tie auth-v2 implementation to CONSISTENT_TOPOLOGY_CHANGES feature`

needs to be squashed